### ISSUE TITLE
Final new tabs not responsive

### ISSUE THREAD

**Issue by JasonSills** [DS4PS/cpp-526-spr-2020#36]
Hi,

For some reason when I create a new tab only one part of the dashboard is responsive. For example, if I add substance to the first tab everything works fine. However, if I copy and paste the same exact same information to create a new tab the filters have no impact on the output. Same exact code copy and pasted in a new tab. First tab works fine, second tab isn't dynamic. 

Traffic Accidents By Day and Time
=====================================  

Inputs {.sidebar}
-------------------------------------
```{r}
checkboxGroupInput("days", label = h3("Day of Week"), 
    choices = list("Monday"    = "Mon", 
                   "Tuesday"   = "Tue", 
                   "Wednesday" = "Wed", 
                   "Thursday"  = "Thu",
                   "Friday"    = "Fri",
                   "Saturday"  = "Sat",
                   "Sunday"    = "Sun" ),
    selected = c("Fri","Sat","Sun"))

sliderInput("hour", label = h3("Time of Day"), 
            min = 0, max = 23, value = c(6, 12))

checkboxGroupInput("d1.subabuse", 
                   label = h3("Driver 1 Influence of Alcohol and Drugs"), 
    choices = list("No Apparent Influence"="No Apparent Influence", 
                   "Alcohol Influence"="Alcohol",
                   "Drug Influence"="Drugs",
                   "Alcohol and  Influence"="Alcohol and Drugs" ),
    selected = c("No Apparent Influence"))

checkboxGroupInput("d2.subabuse", 
                   label = h3("Driver 2 Influence of Alcohol and Drugs"), 
    choices = list("No Apparent Influence"="No Apparent Influence", 
                   "Alcohol Influence"="Alcohol",
                   "Drug Influence"="Drugs",
                   "Alcohol and  Influence"="Alcohol and Drugs" ),
    selected = c("No Apparent Influence"))

```

Outputs
-------------------------------------

### Traffic Accidents By Day and Time

```{r}
renderLeaflet({
  
  days.of.week <- input$days    # vector will all checked values
  start.time <- input$hour[1]   # sliderInput lower value
  end.time  <-  input$hour[2] 
  
  d2 <-
    dat %>%
    filter( day %in% input$days, 
            hour >= start.time & hour <= end.time,
            d1.substance %in% input$d1.subabuse,
            d2.substance %in% input$d2.subabuse)
  
  d2$col.vec <- ifelse( d2$nohurt, "gray20", ifelse(d2$inj, "steelblue", "darkorange") )              
    
  point.size <- d2$Totalinjuries + d2$Totalfatalities

  crash.details <- paste0( "Time: ", d2$DateTime, "<br>",
                     "Total Fatalities: ", d2$Totalfatalities, "<br>",
                     "Total Injuries: ", d2$Totalinjuries, "<br>",
                     "Collision type: ", d2$Collisionmanner)
  
  tempe <- leaflet( ) %>% 
              addProviderTiles( "CartoDB.Positron" )  %>%
              setView( lng=-111.9278, lat=33.39951, zoom=13 )
  
  
  addCircles( tempe, lng=d2$Longitude, lat=d2$Latitude,
              fillColor=d2$col.vec, fillOpacity=0.5, 
              stroke=F, radius=50*(1+0.33*point.size),
              popup=crash.details )


})
```

**Comment by jamisoncrawford** [comment]
What happens when you change the code in the second/pasted tab (*mutatis mutandis*)?

**Comment by JasonSills** [comment]
The graph doesn't work. I deleted all of the code that I changed, but the graph doesn't behave as the original. But, if I just copy and paste the original it works as it should.

**Comment by jamisoncrawford** [comment]
> However, if I copy and paste the same exact same information to create a new tab the filters have no impact on the output. Same exact code copy and pasted in a new tab. First tab works fine, second tab isn't dynamic.

> But, if I just copy and paste the original it works as it should.

Sorry, @JasonSills I'm having some trouble understanding.  Can you send me the .Rmd file via Canvas with the intended new tab/changes?

**Comment by jamisoncrawford** [comment]
For those finding this now - @JasonSills and I were able to troubleshoot this successfully together.

**The Issue:** By using the same `inputId =`, albeit with different code and in a different tab, the catch-all `input$x` (`x` being the ID) was the root of this issue.

Essentially, we needed to create a new, unique value for `inputID =`, even if the calls to input IDs `days` and `hours` were functionally the same.  We used `days2` and `hours2` and it worked!

**Comment by JasonSills** [comment]
Hi, 

I just submitted my dashboard in Canvas, but I tried to get a token in RStudio, but it won't create one.  I click create and "Invalid Token" appears on the bottom right corner of my screen. Worse, the dashboard we worked on yesterday will not work. It broke after I added rsconnect. One of my tabs won't appear now and another has is split down the middle.

**Comment by jamisoncrawford** [comment]
No worries, take an extra 24 hours to see if you can troubleshoot it and
revert it to the original. Will that help?

**Comment by JasonSills** [comment]
Do you remember the one you sent me with edits? That worked fine yesterday, but when I open it it doesn't work. The second tab is broken and one tab will not open. Can you check the one I submitted? I have to work on my final project for the other Program Evaluation class and cannot troubleshoot. Could I look at it on Wednesday?

**Comment by jamisoncrawford** [comment]
Sure, send me your file in the meantime and ill look at it tomorrow. All
will be well, good luck with your other work!

**Comment by JasonSills** [comment]
So I actually just took an older code from yesterday and made adjustments (changing the inputs we discovered yesterday). I have submitted that one on Canvas. Can you please download and make sure it works?

**Comment by jamisoncrawford** [comment]
Yes. I'm in bed now but I will do it first thing tomorrow. Don't worry
about deadline. At any rate, I saw the product working and as I said you'll
get full credit, which you earned and demonstrated.

**Comment by JasonSills** [comment]
Have a good night sleep and rest assured that I think the R Gods have decided to stop mocking me (for now). I think I got it all to work. Please let me know how it looks tomorrow.

**Comment by jamisoncrawford** [comment]
Thanks Jason, will do! And congrats on getting this submitted :).
